### PR TITLE
Bring in BuffersExtensions WriteMultiSegment

### PR DESCRIPTION
**Pull request by benaadams**

### CODE REVIEW

**Comment by davidfowl** [comment]
When I tired this, nothing got faster.

**Comment by benaadams** [comment]
> When I tired this, nothing got faster.

Spoliers... am still running benchmarks

**Comment by benaadams** [comment]
Yeah no real change, wonder about skipping GetSpan

image

**Comment by benaadams** [comment]
Still not a great difference :(

File: src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs [hunk 1]
```diff
@@ -286,31 +286,38 @@ private void WriteMessageCore(HubMessage message, Stream output)
             // PackerCompatibilityOptions.None prevents from serializing byte[] as strings
             // and allows extended objects
             var packer = Packer.Create(output, PackerCompatibilityOptions.None);
-            switch (message)
+
+            if (message is InvocationMessage invocationMessage)
```
[the remark this answers is not in the document]
Comment (benaadams) [reply src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs]: "Pattern matching switch generates junkier code than if" https://github.com/aspnet/SignalR/pull/1918/commits/f9ce4365e30987de562a485d6c07357de50503ac
```
; Total bytes of code 657, prolog size 10 for method
MessagePackHubProtocol:WriteMessageCore(ref,ref):this

vs

; Total bytes of code 612, prolog size 10 for method
MessagePackHubProtocol:WriteMessageCore(ref,ref):this

Still seems noisy though
```

[the remark this answers is not in the document]
Comment (benaadams) [reply src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs]: They both generate junky code; I think so the stack trace unwinds through the correct point :-/

**Comment by benaadams** [comment]
asm is better; but the benchmarks are saying its underwhelming :(

**Pull request closed without merging**